components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.email/src/main/java/org/wso2/carbon/event/output/adapter/email/EmailEventAdapter.java

```diff
@@ -95,22 +116,123 @@ public void testConnect() throws TestConnectionNotSupportedException {
+             */
+
+            final String smtpUsername;
+            final String smtpPassword;
+
+
+            // initialize SMTP session.
+            Properties props = new Properties();
+            props.putAll(globalProperties);
+
+            //Verifying default SMTP properties of the SMTP server.
+
+            smtpFrom = props.getProperty(MailConstants.MAIL_SMTP_FROM);
+            smtpHost = props.getProperty(EmailEventAdapterConstants.MAIL_SMTP_HOST);
+            smtpPort = props.getProperty(EmailEventAdapterConstants.MAIL_SMTP_PORT);
+
+            if (smtpFrom == null) {
+                String msg = "failed to connect to the mail server due to null smtpFrom value";
+                throw new ConnectionUnavailableException("The adapter " +
+                        eventAdapterConfiguration.getName() + " " + msg);
+
+            }
+
+            if (smtpHost == null) {
+                String msg = "failed to connect to the mail server due to null smtpHost value";
+                throw new ConnectionUnavailableException
+                        ("The adapter " + eventAdapterConfiguration.getName() + " " + msg);
+            }
+
+            if (smtpPort == null) {
+                String msg = "failed to connect to the mail server due to null smtpPort value";
+                throw new ConnectionUnavailableException
+                        ("The adapter " + eventAdapterConfiguration.getName() + " " + msg);
+            }
+
+
+            try {
+                smtpFromAddress = new InternetAddress(smtpFrom);
+            } catch (AddressException e) {
+                log.error("Error in retrieving smtp address : " +
+                        smtpFrom, e);
+                String msg = "failed to connect to the mail server due to error in retrieving " +
+                        "smtp from address";
+                throw new ConnectionUnavailableException
+                        ("The adapter " + eventAdapterConfiguration.getName() + " " + msg, e);
+
+            }
+
+            //Retrieving username and password of SMTP server.
+            smtpUsername = props.getProperty(MailConstants.MAIL_SMTP_USERNAME);
+            smtpPassword = props.getProperty(MailConstants.MAIL_SMTP_PASSWORD);
+
+
+            //initializing SMTP server to create session object.
+
+            if (smtpUsername != null && smtpPassword != null) {
+
+                session = Session.getInstance(props, new Authenticator() {
+                    public PasswordAuthentication getPasswordAuthentication() {
+                        return new PasswordAuthentication(smtpUsername, smtpPassword);
+                    }
+                });
+            } else {
+                log.error("Error in smtp username & password verification");
+                String msg = "failed to connect to the mail server due to failed " +
+                        "user password authorization";
+                throw new ConnectionUnavailableException("The adapter " +
```

Comment:
Is there any specific reason for doing both logging the exception and throwing ?